target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vrv9510kwac23.dts

```diff
@@ -0,0 +1,286 @@
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		rfkill {
+			label = "rfkill";
+			gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RFKILL>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		// from top to bottom
+		power_green: power {
+			label = "green:power";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			default-state = "keep";
+		};
+		internet_green: internet_green {
+			label = "green:internet";
+			gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
+		};
+		internet_red {
+			label = "red:internet";
+			gpios = <&gpio 33 GPIO_ACTIVE_LOW>;
+		};
+		phone {
+			label = "green:phone";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+		lan {
+			label = "green:lan";
+			gpios = <&gpio 45 GPIO_ACTIVE_LOW>;
+		};
+		wireless_blue {
+			label = "blue:wlan";
+			gpios = <&gpio 40 GPIO_ACTIVE_LOW>;
+		};
+		wireless_green: wireless_green {
+			label = "green:wlan";
+			gpios = <&gpio 47 GPIO_ACTIVE_LOW>;
+		};
+		upgrade: upgrade {
+			label = "blue:upgrade";
+			gpios = <&gpio 41 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	usb_vbus: regulator-usb-vbus {
+		compatible = "regulator-fixed";
+
+		regulator-name = "USB_VBUS";
+
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+
+		gpio = <&gpio 32 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+};
+
+&localbus {
+	flash@0 {
+		compatible = "lantiq,nand-xway";
+		lantiq,cs = <1>;
+		bank-width = <2>;
+		reg = <0 0x0 0x2000000>;
+
+		pinctrl-0 = <&nand_pins>, <&nand_cs1_pins>;
+		pinctrl-names = "default";
+
+		nand-use-soft-ecc-engine;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "uboot";
+				reg = <0x00000 0x80000>;
+			};
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x80000 0x20000>;
+			};
+			partition@60000 {
+				label = "kernel";
+				reg = <0xA0000 0x3A0000>;
+			};
+			partition@360000 {
+				label = "ubi";
+				reg = <0x3A0000 0x7c60000>;
+			};
+		};
+	};
+};
+
+&gphy0 {
+	lantiq,gphy-mode = <GPHY_MODE_GE>;
+};
+
+&gphy1 {
+	lantiq,gphy-mode = <GPHY_MODE_GE>;
+};
+
+&gpio {
+	pinctrl-names = "default";
+	pinctrl-0 = <&state_default>;
+
+	state_default: pinmux {
+		exin3 {
+			lantiq,groups = "exin3";
+			lantiq,function = "exin";
+		};
+		pci_rst {
+			lantiq,pins = "io21";
+			lantiq,output = <1>;
+			lantiq,open-drain = <0>;
+			lantiq,pull = <2>;
+		};
+		pcie-rst {
+			lantiq,pins = "io38";
+			lantiq,pull = <0>;
+			lantiq,output = <1>;
+		};
+	};
+};
+
+&gswip {
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+};
+
+&gswip_mdio {
+	phy0: ethernet-phy@0 {
+		reg = <0x0>;
+	};
+	phy1: ethernet-phy@1 {
+		reg = <0x1>;
+	};
+	phy5: ethernet-phy@5 {
+		reg = <0x5>;
+	};
+	phy11: ethernet-phy@11 {
```

Comment:
I know this is matches other lantiq devices, but shouldn't this be `phy17: ethernet-phy@11 {`?

Reply:
I don't understand how do you come to that conclusion. Those values are based on the specified for ethernet switch on the VRX200 SoC https://www.kernel.org/doc/Documentation/devicetree/bindings/net/dsa/lantiq-gswip.txt

Reply:
> I don't understand how do you come to that conclusion. Those values are based on the specified for ethernet switch on the VRX200 SoC https://www.kernel.org/doc/Documentation/devicetree/bindings/net/dsa/lantiq-gswip.txt

Yes, but `ethernet-phy@11` means phy at address 0x11, which matches `reg = <0x11>;` (these values are in hex).
However, the name usually contains the value in decimal (0x11 = 17).
Anyway, this is not important, so let's forget about these two and focus on the other comments...

Reply:
I just followed the style I saw on other dts

```
./vr9_arcadyan_vg3503j.dts:72:  phy11: ethernet-phy@11 {
./vr9_arcadyan_vg3503j.dts:79:  phy13: ethernet-phy@13 {
./vr9_avm_fritz736x.dtsi:97:    phy0: ethernet-phy@0 {
./vr9_avm_fritz736x.dtsi:101:   phy1: ethernet-phy@1 {
./vr9_avm_fritz736x.dtsi:105:   phy11: ethernet-phy@11 {
./vr9_avm_fritz736x.dtsi:109:   phy13: ethernet-phy@13 {
./vr9_buffalo_wbmr-300hpd.dts:183:      phy11: ethernet-phy@11 {
./vr9_buffalo_wbmr-300hpd.dts:186:      phy12: ethernet-phy@12 {
./vr9_buffalo_wbmr-300hpd.dts:189:      phy13: ethernet-phy@13 {
./vr9_buffalo_wbmr-300hpd.dts:192:      phy14: ethernet-phy@14 {
./ar9_avm_fritz7320.dts:110:            phy0: ethernet-phy@0 {
./vr9_arcadyan_vgv7510kw22.dtsi:145:    phy1: ethernet-phy@1 {
./vr9_arcadyan_vgv7510kw22.dtsi:148:    phy11: ethernet-phy@11 {
./vr9_arcadyan_vgv7510kw22.dtsi:151:    phy12: ethernet-phy@12 {
./vr9_arcadyan_vgv7510kw22.dtsi:154:    phy13: ethernet-phy@13 {
./vr9_arcadyan_vgv7510kw22.dtsi:157:    phy14: ethernet-phy@14 {
./vr9_arcadyan_arv7519rw22.dts:128:     phy0: ethernet-phy@0 {
./vr9_arcadyan_arv7519rw22.dts:131:     phy11: ethernet-phy@11 {
./vr9_arcadyan_arv7519rw22.dts:134:     phy12: ethernet-phy@12 {
./vr9_arcadyan_arv7519rw22.dts:137:     phy13: ethernet-phy@13 {
./vr9_arcadyan_arv7519rw22.dts:140:     phy14: ethernet-phy@14 {
./ar9_avm_fritz7312.dts:108:            phy0: ethernet-phy@0 {
./vr9_netgear_dm200.dts:91:     phy13: ethernet-phy@13 {
./vr9_alphanetworks_asl56026.dts:81:    phy11: ethernet-phy@11 {
./vr9_alphanetworks_asl56026.dts:85:    phy14: ethernet-phy@14 {
./vr9_tplink_tdw89x0.dtsi:150:  phy0: ethernet-phy@0 {
./vr9_tplink_tdw89x0.dtsi:154:  phy5: ethernet-phy@5 {
./vr9_tplink_tdw89x0.dtsi:157:  phy11: ethernet-phy@11 {
./vr9_tplink_tdw89x0.dtsi:160:  phy13: ethernet-phy@13 {
./vr9_avm_fritz7430.dts:204:    phy11: ethernet-phy@11 {
./vr9_avm_fritz7430.dts:207:    phy12: ethernet-phy@12 {
./vr9_avm_fritz7430.dts:210:    phy13: ethernet-phy@13 {
./vr9_avm_fritz7430.dts:213:    phy14: ethernet-phy@14 {
./vr9_bt_homehub-v5a.dts:156:   phy0: ethernet-phy@0 {
./vr9_bt_homehub-v5a.dts:159:   phy1: ethernet-phy@1 {
./vr9_bt_homehub-v5a.dts:162:   phy5: ethernet-phy@5 {
./vr9_bt_homehub-v5a.dts:165:   phy11: ethernet-phy@11 {
./vr9_bt_homehub-v5a.dts:168:   phy13: ethernet-phy@13 {
./vr9_lantiq_easy80920.dtsi:142:        phy0: ethernet-phy@0 {
./vr9_lantiq_easy80920.dtsi:145:        phy1: ethernet-phy@1 {
./vr9_lantiq_easy80920.dtsi:148:        phy5: ethernet-phy@5 {
./vr9_lantiq_easy80920.dtsi:151:        phy11: ethernet-phy@11 {
./vr9_lantiq_easy80920.dtsi:154:        phy13: ethernet-phy@13 {
./vr9_tplink_vr200.dtsi:140:    phy0: ethernet-phy@0 {
./vr9_tplink_vr200.dtsi:144:    phy5: ethernet-phy@5 {
./vr9_tplink_vr200.dtsi:147:    phy11: ethernet-phy@11 {
./vr9_tplink_vr200.dtsi:150:    phy13: ethernet-phy@13 {
./vr9_arcadyan_vrv9510kwac23.dts:203:   phy0: ethernet-phy@0 {
./vr9_arcadyan_vrv9510kwac23.dts:206:   phy1: ethernet-phy@1 {
./vr9_arcadyan_vrv9510kwac23.dts:209:   phy5: ethernet-phy@5 {
./vr9_arcadyan_vrv9510kwac23.dts:212:   phy11: ethernet-phy@11 {
./vr9_arcadyan_vrv9510kwac23.dts:215:   phy13: ethernet-phy@13 {
./vr9_avm_fritz3370-rev2.dtsi:151:      phy0: ethernet-phy@0 {
./vr9_avm_fritz3370-rev2.dtsi:156:      phy1: ethernet-phy@1 {
./vr9_avm_fritz3370-rev2.dtsi:161:      phy11: ethernet-phy@11 {
./vr9_avm_fritz3370-rev2.dtsi:165:      phy13: ethernet-phy@13 {
./vr9_avm_fritz7412.dts:187:    phy11: ethernet-phy@11 {
./vr9_arcadyan_vgv7519.dtsi:163:        phy0: ethernet-phy@0 {
./vr9_arcadyan_vgv7519.dtsi:166:        phy1: ethernet-phy@1 {
./vr9_arcadyan_vgv7519.dtsi:169:        phy5: ethernet-phy@5 {
./vr9_arcadyan_vgv7519.dtsi:172:        phy11: ethernet-phy@11 {
./vr9_arcadyan_vgv7519.dtsi:175:        phy13: ethernet-phy@13 {
./vr9_zyxel_p-2812hnu-fx.dtsi:158:      phy0: ethernet-phy@0 {
./vr9_zyxel_p-2812hnu-fx.dtsi:161:      phy1: ethernet-phy@1 {
./vr9_zyxel_p-2812hnu-fx.dtsi:164:      phy5: ethernet-phy@5 {
./vr9_zyxel_p-2812hnu-fx.dtsi:167:      phy11: ethernet-phy@11 {
./vr9_zyxel_p-2812hnu-fx.dtsi:170:      phy13: ethernet-phy@13 {
./vr9_avm_fritz3390.dts:150:    phy0: ethernet-phy@0 {
./vr9_avm_fritz3390.dts:155:    phy1: ethernet-phy@1 {
./vr9_avm_fritz3390.dts:160:    phy11: ethernet-phy@11 {
./vr9_avm_fritz3390.dts:164:    phy13: ethernet-phy@13 {
```
Looks like they all match the hexadecimal value for the phy id